src/Models/IoTHubDevice.ts

```diff
@@ -138,12 +197,23 @@ export class IoTHubDevice implements Component, Provisionable {
     return true;
   }
 
-  updateConfigSettings(
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    _type: ScaffoldType,
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    _componentInfo?: ComponentInfo
-  ): void {
-    // Do nothing.
+  async updateConfigSettings(type: ScaffoldType, componentInfo?: ComponentInfo): Promise<void> {
+    const iotHubComponentIndex = await this.azureConfigFileHandler.getComponentIndexById(type, this.id);
+    if (iotHubComponentIndex > -1) {
+      if (!componentInfo) {
+        return;
+      }
+      await this.azureConfigFileHandler.updateComponent(type, iotHubComponentIndex, componentInfo);
+    } else {
+      const newIotHubConfig: AzureComponentConfig = {
+        id: this.id,
+        folder: "",
+        name: "",
+        dependencies: this.dependencies,
+        type: this.componentType,
+        componentInfo

```

Comment:
what if componentInfo is undefined here? Do we need to check componentInfo first?

Reply:
It's allowed that if componentInfo is undefined. "appendComponent" function takes component as param, it's good as long as component is not undefined.